models/pull.go

```diff
@@ -556,6 +556,8 @@ func (pr *PullRequest) Merge(doer *User, baseGitRepo *git.Repository, mergeStyle
 	}
 
 	env := PushingEnvironment(doer, pr.BaseRepo)
+	// FIXME: #6946 requires protected branches to be ignored
+	env = env[:len(env)-1]
```

Comment:
This will assume that env is the last one. It's danger. I would like to traverse all the envs and remove the extract one.

Reply:
It's not a longstanding fix it's just meant to last until I get #6961 fixed.